sbt-dotty/src/dotty/tools/sbtplugin/DottyPlugin.scala

```diff
@@ -167,6 +201,15 @@ object DottyPlugin extends AutoPlugin {
           (incOptions in Compile).value
       },
 
+      scalacOptions := {
+        val options = scalacOptions.value
+        if (autoCompilerPlugins.value) {
+          val newPlugins = autoPlugins(update.value, internalPluginClasspath.value.files)
+          val existing = options.toSet
+          options ++ newPlugins.filterNot(existing)
+        } else options
+      },
+
```

Comment:
@smarter : could you please advise if it's the right thing to do here?

Reply:
The right thing to do is to fix sbt upstream to look for plugin.properties. Meanwhile, we can use whatever hack happens to work.

Reply:
Thanks a lot, I'll make an upstream PR.